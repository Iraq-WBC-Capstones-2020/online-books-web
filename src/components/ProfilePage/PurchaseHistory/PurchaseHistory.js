import React, { useState } from 'react';
import { MDBDataTableV5 } from 'mdbreact';
import { useTranslation } from 'react-i18next';

function PurchaseHistory() {
  const { t } = useTranslation();
  const SETS_ENTRIES_OPTION_TO_DISPALY = [5, 10, 20];
  const INITIAL_NUMBER_PAGES_PAGINATION = 4;
  const NUMBERS_OF_ROWS = 5;
  const numberOfPurchases = 23;
  const purchases = Array(numberOfPurchases).fill({
    orderId: '25565845',
    books: 'System Architect',
    status: 'Edinburgh',
    originalPrice: '61',
    totalPrice: '2011/04/25',
  });
  const [datatable] = useState({
    columns: [
      {
        label: t('profilePage.orderId'),
        field: 'orderId',
        attributes: {
          'aria-controls': 'DataTable',
          'aria-label': 'orderId',
        },
      },
      {
        label: t('profilePage.books'),
        field: 'books',
      },
      {
        label: t('profilePage.status'),
        field: 'status',
      },
      {
        label: t('profilePage.originalPrice'),
        field: 'originalPrice',
      },
      {
        label: t('profilePage.totalPrice'),
        field: 'totalPrice',
      },
    ],
    rows: purchases,
  });
  return (
    <MDBDataTableV5
      hover
      entries={NUMBERS_OF_ROWS}
      entriesOptions={SETS_ENTRIES_OPTION_TO_DISPALY}
      pagesAmount={INITIAL_NUMBER_PAGES_PAGINATION}
      data={datatable}
    />
  );
}

export default PurchaseHistory;
