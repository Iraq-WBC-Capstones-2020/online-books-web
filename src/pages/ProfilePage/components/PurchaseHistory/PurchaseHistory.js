import React, { useState } from 'react';
import { MDBDataTableV5 } from 'mdbreact';
import { useTranslation } from 'react-i18next';

function PurchaseHistory() {
  const { t } = useTranslation();

  const [datatable] = useState({
    columns: [
      {
        label: t('profilePage.orderId'),
        field: 'orderId',
        attributes: {
          'aria-controls': 'DataTable',
          'aria-label': 'orderId',
        },
      },
      {
        label: t('profilePage.books'),
        field: 'books',
      },
      {
        label: t('profilePage.status'),
        field: 'status',
      },
      {
        label: t('profilePage.originalPrice'),
        field: 'originalPrice',
      },
      {
        label: t('profilePage.totalPrice'),
        field: 'totalPrice',
      },
    ],
    rows: [
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
      {
        orderId: 'Tiger Nixon',
        books: 'System Architect',
        status: 'Edinburgh',
        originalPrice: '61',
        totalPrice: '2011/04/25',
      },
    ],
  });
  return (
    <MDBDataTableV5
      hover
      entriesOptions={[5, 20, 25]}
      entries={5}
      pagesAmount={4}
      data={datatable}
    />
  );
}

export default PurchaseHistory;
