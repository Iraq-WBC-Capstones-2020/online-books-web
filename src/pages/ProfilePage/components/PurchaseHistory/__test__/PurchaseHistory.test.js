import React from 'react';
import renderer from 'react-test-renderer';
import PurchaseHistory from '../PurchaseHistory';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
test('PurchaseHistory snapshot', () => {
  jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key) => key }),
  }));
  const history = createMemoryHistory();

  const tree = renderer
    .create(
      <Router history={history}>
        <PurchaseHistory />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
