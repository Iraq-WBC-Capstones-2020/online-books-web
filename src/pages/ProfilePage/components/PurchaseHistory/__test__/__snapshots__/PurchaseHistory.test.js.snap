// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseHistory snapshot 1`] = `
<div
  className="mdb-datatable dt-bootstrap4"
  data-test="datatable"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
      data-test="mdb-datatable-table"
    >
      <div
        className=""
        data-test="table"
        style={
          Object {
            "maxHeight": undefined,
          }
        }
      >
        <table
          className="table table-borderless table-hover mdb-dataTable"
          entries={5}
        >
          <thead
            className="mdb-dataTable-head"
            data-test="datatable-head"
          >
            <tr>
              <th
                aria-controls="DataTable"
                aria-label="orderId"
                className="sorting"
                onClick={[Function]}
              >
                profilePage.orderId
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                profilePage.books
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                profilePage.status
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                profilePage.originalPrice
              </th>
              <th
                className="sorting"
                onClick={[Function]}
              >
                profilePage.totalPrice
              </th>
            </tr>
          </thead>
          <tbody
            data-test="table-body"
          >
            <tr>
              <td>
                25565845
              </td>
              <td>
                System Architect
              </td>
              <td>
                Edinburgh
              </td>
              <td>
                61
              </td>
              <td>
                2011/04/25
              </td>
            </tr>
            <tr>
              <td>
                25565845
              </td>
              <td>
                System Architect
              </td>
              <td>
                Edinburgh
              </td>
              <td>
                61
              </td>
              <td>
                2011/04/25
              </td>
            </tr>
            <tr>
              <td>
                25565845
              </td>
              <td>
                System Architect
              </td>
              <td>
                Edinburgh
              </td>
              <td>
                61
              </td>
              <td>
                2011/04/25
              </td>
            </tr>
            <tr>
              <td>
                25565845
              </td>
              <td>
                System Architect
              </td>
              <td>
                Edinburgh
              </td>
              <td>
                61
              </td>
              <td>
                2011/04/25
              </td>
            </tr>
            <tr>
              <td>
                25565845
              </td>
              <td>
                System Architect
              </td>
              <td>
                Edinburgh
              </td>
              <td>
                61
              </td>
              <td>
                2011/04/25
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="d-flex w-100 justify-content-between"
    >
      <div
        className="row ml-3 justify-content-between"
      >
        <div
          className="false"
          data-test="datatable-search"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <div
            className="mdb-datatable-filter flex-row"
            data-test="datatable-input"
          >
            <input
              aria-label="Search"
              className="form-control form-control-sm ml-0 my-1"
              onChange={[Function]}
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="d-flex align-items-center justify-content-end"
      >
        <div
          className="mdb-datatable-entries"
          data-test="mdb-datatable-entries"
        >
          <div
            className="mdb-datatable-length bs-select"
            data-test="datatable-select"
          >
            <label>
              Rows per page:
              <select
                className="custom-select custom-select-sm form-control form-control-sm"
                onChange={[Function]}
                style={
                  Object {
                    "marginLeft": ".5rem",
                  }
                }
                value={5}
              >
                <option
                  value={5}
                >
                  5
                </option>
                <option
                  value={10}
                >
                  10
                </option>
                <option
                  value={20}
                >
                  20
                </option>
              </select>
            </label>
          </div>
        </div>
        <div
          className="mdb-datatable-info d-flex align-items-center"
          data-test="datatable-info"
        >
          <div
            aria-live="polite"
            role="status"
          >
             1 - 5 of 23 
          </div>
        </div>
        <div
          data-test="datatable-pagination"
        >
          <div
            className="mdb-dataTables_paginate"
          >
            <ul
              className="pagination m-0"
              data-test="pagination"
            >
              <li
                className="disabled page-item"
                data-test="page-item"
              >
                <a
                  aria-label="Prev"
                  className="page-link page-link"
                  data-test="page-link"
                  onClick={[Function]}
                >
                  <span>
                    <i
                      className="fas fa-chevron-left"
                    />
                  </span>
                </a>
              </li>
              <li
                className="page-item"
                data-test="page-item"
              >
                <a
                  aria-label="Next"
                  className="page-link page-link"
                  data-test="page-link"
                  onClick={[Function]}
                >
                  <span>
                    <i
                      className="fas fa-chevron-right"
                    />
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
